src/hats/pixel_math/partition_stats.py

```diff
@@ -113,21 +122,42 @@ def generate_alignment(
         if the histogram is the wrong size, or some initial histogram bins
         exceed threshold.
     """
-    _validate_alignment_arguments(histogram, highest_order, lowest_order, threshold)
+    # Validate inputs.
+    _validate_alignment_arguments(
+        row_count_histogram, mem_size_histogram, highest_order, lowest_order, threshold
+    )
 
-    nested_sums = _get_nested_sums(histogram, highest_order, lowest_order)
+    # Generate nested sums.
+    nested_sums_row_count = _get_nested_sums(row_count_histogram, highest_order, lowest_order)
+    if mem_size_histogram is not None:
+        nested_sums_mem_size = _get_nested_sums(mem_size_histogram, highest_order, lowest_order)
+    else:
+        nested_sums_mem_size = None
 
+    # Generate alignment.
     if drop_empty_siblings:
-        return _get_alignment_dropping_siblings(nested_sums, highest_order, lowest_order, threshold)
-    return _get_alignment(nested_sums, highest_order, lowest_order, threshold)
+        return _get_alignment_dropping_siblings(
+            nested_sums_row_count, highest_order, lowest_order, threshold, nested_sums_mem_size
+        )
+    return _get_alignment(nested_sums_row_count, highest_order, lowest_order, threshold, nested_sums_mem_size)
```

Comment:
How about, instead:

```
alignment = _get_alignment(...)
if mem_size_histogram is not None:
  // transform to the 3-tuple
  alignment = [[thing.0, thing.1, thing.3] for thing in alignment]
  return alignment
return alignment
```

That simplifies the import code, as you're never using the mem_size sum later. And it keeps this method true to the docstring that says it returns a 3-tuple